core/app/models/spree/order.rb

```diff
@@ -138,9 +138,9 @@ def states
     accepts_nested_attributes_for :shipments
 
     # Needs to happen before save_permalink is called
-    before_validation :set_currency
-    before_validation :clone_billing_address, if: :use_billing?
     before_validation :ensure_store_presence
+    before_validation :set_currency
```

Comment:
@VygovskySergey let's rename it to `ensure_currency` to keep up with the convention

Let's add `ensure_currency` method and use it in `set_currency` (with a deprecation warning)

Reply:
In which version will we delete `set_currency`?

Reply:
@VygovskySergey let's do it in `4.0`, what do you think?

Reply:
I think that 4.0 will be major release, maybe 3.9?